snoop/data/management/commands/filestats.py

```diff
@@ -0,0 +1,181 @@
+
+
+def get_top_extensions(collections_list, print_supported=True):
+    """Return a dictionary of file extensions that occupy most space in collections.
+
+    Args:
+        collections_list: A list of collections that will be analyzed.
+        print_supported: When False only analyzes unsupported filetypes.
+    """
+    ext_dict = {}
+    for col in collections_list:
+        query = r"""select substring(encode(f.name_bytes::bytea, 'escape')::text
+                    from '(\..{1,20})$') as ext,
+                    sum(f.size) as size,
+                    b.mime_type as mime
+                    from data_file f
+                    join data_blob b on f.blob_id = b.sha3_256
+                    group by ext, mime
+                    order by size desc limit 100;"""
+        with connections[collections.ALL[col].db_alias].cursor() as cursor:
+            cursor.execute(query)
+            results = cursor.fetchall()
+
+        for ext, size, mime in results:
+            if not print_supported:
+                if mime in SUPPORTED_MIME_TYPES:
+                    continue
+            if ext not in ext_dict:
+                ext_dict[ext] = {'size': truncate_size(int(size)), 'mtype': set([mime])}
+            else:
+                ext_dict[ext]['size'] += truncate_size(int(size))
+                ext_dict[ext]['mtype'].add(mime)
+    sorted_ext_dict = sorted(ext_dict.items(), key=lambda x: x[1]['size'], reverse=True)
+    return dict(sorted_ext_dict)
+
+
+def get_description(col, mime_type, *extension):
+    """Return the magic description for a given mime-type.
+
+    Args:
+        col: Collection on which the query is executed.
+        mime_type: Mime-Type for which the descriptions is returned.
+        *extension: Optional file-extension as string to limit the search to this
+            extension.
+    """
+
+    collection = collections.ALL[col]
+    with collection.set_current():
+        try:
+            if extension:
+                queryset = models.File.objects\
+                    .annotate(str_name=RawSQL("encode(name_bytes::bytea, 'escape')::text", ()))\
+                    .filter(blob__mime_type=mime_type, str_name__endswith=extension[0])\
+                    .values("blob__magic")[0]
+            else:
+                extension = [""]
+                queryset = models.File.objects\
+                    .annotate(str_name=RawSQL("encode(name_bytes::bytea, 'escape')::text", ()))\
+                    .filter(blob__mime_type=mime_type)\
+                    .values("blob__magic")[0]
+        except IndexError:
+            return None
+    return queryset['blob__magic']
+
+
+class Command(BaseCommand):
+    """Print the statistics for mimetypes or file-extendsion"""
+    help = "Display filetype stats."
+
+    def add_arguments(self, parser):
+        """Arguments to show only unsupported types, include magic descriptions,
+        include full magic descriptions and for choosing specific collections"""
+
```

Comment:
Add a new numeric argument called `--row-count` default 100 that you'll use to limit the number of rows in each request.